test/integration/targets/lineinfile/tasks/test_literal.yml

```diff
@@ -0,0 +1,318 @@
+      register: result
+
+    - name: assert that the line was removed
+      assert:
+        that:
+          - result is changed
+          - "result.msg == '1 line(s) removed'"
+
+    - name: stat the test after the middle line was removed
+      stat:
+        path: "{{ output_dir }}/testliteral.txt"
+      register: result
+
+    - name: assert test checksum matches after the middle line was removed
+      assert:
+        that:
+          - "result.stat.checksum == '87af6beb4be7ec9f68f7accd2f20367cb6eca934'"
+
+    - name: run a validation script that succeeds using literal
+      lineinfile:
+        dest: "{{ output_dir }}/testliteral.txt"
+        state: absent
+        literal: "This is line 5"
+        validate: "true %s"
+      register: result
+
+    - name: assert that the file validated after removing a line
+      assert:
+        that:
+          - result is changed
+          - "result.msg == '1 line(s) removed'"
+
+    - name: stat the test after the validation succeeded
+      stat:
+        path: "{{ output_dir }}/testliteral.txt"
+      register: result
+
+    - name: assert test checksum matches after the validation succeeded
+      assert:
+        that:
+          - "result.stat.checksum == 'ba9600b34febbc88bfb3ca99cd6b57f1010c19a4'"
+
+    - name: run a validation script that fails using literal
+      lineinfile:
+        dest: "{{ output_dir }}/testliteral.txt"
+        state: absent
+        literal: "This is line 1"
+        validate: "/bin/false %s"
+      register: result
+      ignore_errors: yes
+
+    - name: assert that the validate failed
+      assert:
+        that:
+          - "result.failed == true"
+
+    - name: stat the test after the validation failed
+      stat:
+        path: "{{ output_dir }}/testliteral.txt"
+      register: result
+
+    - name: assert test checksum matches the previous after the validation failed
+      assert:
+        that:
+          - "result.stat.checksum == 'ba9600b34febbc88bfb3ca99cd6b57f1010c19a4'"
+
+  when: stage == "first"
+
+# End of literal tests
+###################################################################
+
+###################################################################
+# 2nd literal tests
+
+- name: Second Literal tests block
+  block:
+
+    - name: Deploy the testliteral.conf file
+      copy:
+        src: testliteral.conf
+        dest: "{{ output_dir }}/testliteral.conf"
+      register: result
+
+    - name: Assert that the testliteral.conf file was deployed
+      assert:
+        that:
+          - result is changed
+          - result.checksum == '6037f13e419b132eb3fd20a89e60c6c87a6add38'
+          - result.state == 'file'
+
+    # Test instertafter
+    - name: Insert lines after with literal
+      lineinfile:
+        path: "{{ output_dir }}/testliteral.conf"
+        literal: "{{ item.regexp }}"
+        line: "{{ item.line }}"
+        insertafter: "{{ item.after }}"
+      with_items: "{{ test_befaf_regexp }}"
```

Comment:
The `line`  and `literal` values for these tests do not contain regexp control characters. I think at least some of them should to adequately test the feature.

Reply:
Added:

```
This is line 1
This is line 2
(\\w)(\\s+)([\\.,])
This is line 4
<FilesMatch ".py[45]?$">
```